### ISSUE TITLE
Some inconsistencies within the html report when using POM

### ISSUE THREAD

**Issue by NikkTod** [vitalets/playwright-bdd#27]
So I have following structure: 
features
steps
pages/sections

In the html report I sometimes see wrong snapshot of the code in that second and third level.

See following examples:

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/b52c774b-8ae2-4e61-b14b-8480be5e6b13)

Also when we have iframes it does not look very nice.

Another example: 
![image](https://github.com/vitalets/playwright-bdd/assets/94455079/191aa623-4ac4-42b7-a528-03616b26670e)

**Comment by NikkTod** [comment]
@vitalets the same applies when using UI mode, it is not giving the correct source line for some of the steps:

![image](https://github.com/vitalets/playwright-bdd/assets/94455079/b798ed39-d57d-4166-a17e-a6ef7168d50b)

**Comment by vitalets** [comment]
Will have a look, thanks for the reporting.
For iframe - could you check does it work correctly for regular playwright tests without bdd generation?

**Comment by vitalets** [comment]
@NikkTod could you please add more details?
1. does these wrong coordinates reproduced only when using iframes?
2. does it work correctly for regular playwright tests without bdd generation?

Thanks in advance.

**Comment by jzaratei** [comment]
Hello @vitalets , I can see the same issue in my reports.
The error is pointing to the wrong line in the page object:

Error should be in line 29:
![image](https://github.com/vitalets/playwright-bdd/assets/47472889/cf328bc0-b104-4c2d-acb6-c830c030621b)

In steps description the error should point to line 29

![image](https://github.com/vitalets/playwright-bdd/assets/47472889/f2496c45-db2a-4439-b9e7-f460adee4e2c)

page object:
![image](https://github.com/vitalets/playwright-bdd/assets/47472889/bbd9fc4c-1876-4066-912c-2ba4c14836f1)

**Comment by vitalets** [comment]
Thanks. Will try to repclicate

**Comment by jzaratei** [comment]
> Thanks. Will try to repclicate. @jzaratei could you expand other locators in your report? Do they also have wrong positions?

Yeah, those marked in red have a wrong position. I was doing some debugging and I found this issue:
When the execution reach the breakpoint in line 31 (my test is failing here), you can see in the line 45 an additional line highlighted in blue and an icon waiting. I don't really  know why that line is marked, but when the report is generated that is the "wrong" line that is display as failed.

![image](https://github.com/vitalets/playwright-bdd/assets/47472889/e73214db-46e5-435e-a2b0-8aa9e1c92bd5)

**Comment by vitalets** [comment]
@jzaratei seems like this is related to ts-node compilation. Could you run your tests with the following command:
```
npx bddgen && NODE_OPTIONS="-r ts-node/register" npx playwright test
```
(keeping `requireModule: ['ts-node/register']` inside playwright config).

For me this shows correct positions for all sub-steps.

**Comment by jzaratei** [comment]
> @jzaratei seems like this is related to ts-node compilation. Could you run your tests with the following command:
> 
> ```
> npx bddgen && NODE_OPTIONS="-r ts-node/register" npx playwright test
> ```
> 
> (keeping `requireModule: ['ts-node/register']` inside playwright config).
> 
> For me this shows correct positions for all sub-steps.

I try to run that command on windows using the cross-env package but no luck... I get this error

The system cannot find the path specified.
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: spawn ts-node\register ENOENT

**Comment by jzaratei** [comment]
@vitalets Excellent work! Now  the error trace are accurate

![Untitled](https://github.com/vitalets/playwright-bdd/assets/47472889/fec603ba-e6af-4b35-9e10-12f73fb74be7)

**Comment by NikkTod** [comment]
@vitalets @jzaratei Working for me as well, sorry for not having the time to debug, busy project days. 

Thanks very much for the fix.

**Comment by vitalets** [comment]
Thank you guys.
As an additional effect I think there is some performance improvement because `ts-node` was replaced by Playwright's built-in loader for TypeScript.

**Comment by vitalets** [comment]
Fixed in v3.3.0. Feel free to reopen if there are any issues.
